manager/sql/manager_sql.go

```diff
@@ -441,6 +441,22 @@ func (s *SQLManager) Get(id string) (Policy, error) {
 	return policies[0], nil
 }
 
+// Update updates an existing policy.
+func (s *SQLManager) Update(policy Policy) (err error) {
+	tx, err := s.db.Begin()
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	if err := s.Delete(policy.GetID()); err != nil {
```

Comment:
I don't think this works like that, the transaction you're rolling back isn't covering the delete command, so basically, on error, nothing happens. Instead we probably need to create a private function that takes a transaction and executes the delete command, and use that private function in update as well as delete.